### DISCUSSION TITLE
Is the comment about maybeRotateStreams valid

### DISCUSSION THREAD

**Discussion by honnix** [prometheus/client_golang#1069] [Q&A]
https://github.com/prometheus/client_golang/blame/main/prometheus/summary.go#L355, this comment says this method requires both `mtx AND bufMtx locked`, but from what I understood, the call site doesn't lock `bufMtx`, for example https://github.com/prometheus/client_golang/blame/main/prometheus/summary.go#L306. Also the comment of `flushColdBuf` doesn't mention it.

**Comment by beorn7** [comment]
Hmm, the method name has also diverged from the name in the doc comment (`maybeRotateStreams` vs. `rotateStreams`). Maybe the comment is just outdated. I wrote this, but it's too long ago. I would assume if we had a race condition here, it would have been detected by now… But of course, a more deductive answer would be great.

**Reply by honnix** [reply]
Yeah it would be good to fix those comments if there is no race condition. I was looking at this code because we had some memory issue, but it could be just we have way too many summaries because of high cardinality.

Thanks for the answer.
